src/query/api/v1/handler/placement/set.go

```diff
@@ -93,41 +93,55 @@ func (h *SetHandler) ServeHTTP(
 		return
 	}
 
+	var isNewPlacement bool
 	curPlacement, err := service.Placement()
-	if err == kv.ErrNotFound {
-		logger.Error("placement not found", zap.Any("req", req), zap.Error(err))
-		xhttp.Error(w, err, http.StatusNotFound)
-		return
-	}
 	if err != nil {
-		logger.Error("unable to get current placement", zap.Error(err))
-		xhttp.Error(w, err, http.StatusInternalServerError)
-		return
+		if err != kv.ErrNotFound {
```

Comment:
it might be nice to document this not-found behavior on the method at least. i'm guessing there isn't api docs for these admin endpoints.

Reply:
There is but certain endpoints aren’t well documented and/or are slightly out of date.

We would like to move to hosting these endpoints behind a GRPC gateway so that the API docs can be auto generated.

For now these are the by hand API docs:
https://docs.m3db.io/openapi/